### PR TITLE
:broom: Cleanup config files

### PR DESCRIPTION
**Pull request by mayorova**
Clean up configuration files:

- Removed obsolete config files that are no longer used:
  - cubert.yml
  - web_analytics.yml
  - message_bus.yml (see https://github.com/3scale/porta/pull/2687)
  - mixpanel.yml
  - cable.yml
  - newrelic.yml
  - storage.yml (we will restore it when migrate to Active Storage)

- Removed `config/docker` folder
  The Dockerfile for the containerized image copies first from `config/docker` and then from `openshift/system/config`, effectively overwriting the former. Unless there is any use case where we need  to build a container image, but NOT for OpenShift environment, we may reconsider. But in any case the configs added by `openshift/system/config` should be flexible enough to cover these use cases with help of environment variables.
Also, the folder `config/examples` contains all possible configuration files, that also permit flexible customization via env vars, and include different environments (development, test, production).

- Removed `preview` from the config files:
  - on OCP we use `production` for both production and staging environments
  - in on-premises only `production` env is used - there is no default mechanism to set the env (only manually adding `RAILS_ENV` env var) and in the current config files there is no difference between `preview` and `production` envs.

- Cleaned up some config files:
  - following the same order for envs: development, test, production
  - using the same name for the "base" config:
    ```
    base: &default
      [...]

    development:
      <<: *default
    ```

P.S. I also had an idea to reorder the keys in the config files in the alphabetical order, but not sure how practical this is. To me it seems better to have them sorted, but if we decide to do so, I'd do it in another PR, otherwise it's impossible to compare the changes.

### CODE REVIEW

File: config/examples/oauth2.yml [hunk 1]
```diff
@@ -1,8 +1,11 @@
-development:
+base: &default
   github:
-    enabled: true
-    client_id: client_id
-    client_secret: client_secret
+    enabled: false
```
Comment (jlledom) [inline config/examples/oauth2.yml:3]: Why disabling it? this way it'll remain disabled even when the env variables are set correctly

File: config/examples/paperclip.yml [hunk 2]
```diff
@@ -11,3 +11,6 @@ development:
 
 test:
   <<: *<%= ENV['FILE_UPLOAD_STORAGE'] || 'filesystem' %>
+
+production:
+  <<: *<%= ENV['FILE_UPLOAD_STORAGE'].presence || 'filesystem' %>
```
Comment (jlledom) [inline config/examples/paperclip.yml:16]: OCD alarm!

File: config/examples/rolling_updates.yml [hunk 3]
```diff
@@ -14,22 +13,17 @@ base: &default
   policy_registry: true
   proxy_private_base_path: true
   service_mesh_integration: false
-  api_as_product: false
+  apicast_v2: true
+  published_service_plan_signup: true
+  apicast_oidc: true
+  provider_sso:
+  billable_contracts:
```
Comment (jlledom) [inline config/examples/rolling_updates.yml:20]: Where did you get this list from? Were all these features implemented but not enabled? I don't see them in the production config in the deploy repo

File: config/examples/segment.yml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 base: &default
-  enabled: true
+  enabled: false
```
Comment (jlledom) [inline config/examples/segment.yml:2]: What is this segment thing for?

File: config/secrets.yml [hunk 5]
```diff
@@ -17,28 +17,19 @@
 
 # Environmental secrets are only available for that specific environment.
 
-default: &base
-  support_provider_key: <%= ENV['SUPPORT_PROVIDER_KEY'] %>
-  support_provider_key_preview: <%= ENV['SUPPORT_PROVIDER_KEY_PREVIEW'] %>
+base: &default
+  secret_key_base: dummy
```
Comment (jlledom) [inline config/secrets.yml:21]: I don't understand this change. First, what were support provider keys for? Second, wouldn't be better to set `secret_key_base: <%= ENV['SECRET_KEY_BASE'] %>` as default? why setting `dummy` as default and then overwrite it only for production?

File: config/examples/paperclip.yml (hunk 2 above)
Comment (akostadinov) [reply config/examples/paperclip.yml:16]: wdym?

File: config/examples/segment.yml (hunk 4 above)
Comment (akostadinov) [reply config/examples/segment.yml:2]: There was some discussion behind it. I forgot though.

File: config/secrets.yml (hunk 5 above)
Comment (akostadinov) [reply config/secrets.yml:21]: We don't really have to set it in fact. This is in [rails](https://github.com/rails/rails/blob/v5.2.7.1/railties/lib/rails/application.rb#L428-L436):
```rb
    def secret_key_base
      if Rails.env.development? || Rails.env.test?
        secrets.secret_key_base ||= generate_development_secret
      else
        validate_secret_key_base(
          ENV["SECRET_KEY_BASE"] || credentials.secret_key_base || secrets.secret_key_base
        )
      end
    end
```

Better to just remove this part.

File: config/examples/paperclip.yml (hunk 2 above)
Comment (mayorova) [reply config/examples/paperclip.yml:16]: @akostadinov It's just that the empty line is missing at the end of the file :upside_down_face:

File: config/secrets.yml (hunk 5 above)
Comment (mayorova) [reply config/secrets.yml:21]: Oh, great point, thanks @akostadinov, I'll remove it.

@jlledom `support_provider_key` was some config that is not used anymore. At some point the swagger specs for 3scale API were stored in "support"'s SaaS account, and the provider key was needed to get those specs. See https://github.com/3scale/system/commit/c3333261bf58699b95087138ba646da585f5ebc8#diff-8dd79c7a3211830fc048071420a263d9f77fa48e0d9d0406b98e61e27cb2d7bf

This is not used anymore.

File: config/examples/rolling_updates.yml (hunk 3 above)
Comment (jlledom) [reply config/examples/rolling_updates.yml:20]: I was looking at the wrong file  🫠

File: config/examples/oauth2.yml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 base: &default
   github:
-    enabled: false
+    enabled: <%= ENV.fetch('GITHUB_ENABLED', true) %>
```
Comment (akostadinov) [inline config/examples/oauth2.yml:3]: wait, how is one supposed to disable this?

**Comment by akostadinov** [comment]
Moving `thinking_sphinx.yml` will require a Dockerfile update for its path.

File: config/examples/redis.yml [hunk 7]
```diff
@@ -1,22 +1,17 @@
...
-  <<: *base
+  <<: *default
 
 test:
-  <<: *base
+  <<: *default
   db: <%= 2 + ENV['TEST_ENV_NUMBER'].to_i %>
+
+production:
+  <<: *default
```
Comment (akostadinov) [inline config/examples/redis.yml:17]: Was it intentional to default production redis to `redis://localhost/1`? I think it makes sense to keep default different from `developement`.

File: config/examples/settings.yml [hunk 8]
```diff
@@ -36,17 +32,24 @@ base: &default
   onpremises_api_docs_version: false
   assets_cdn_host: <%= ENV.fetch('ASSETS_CDN_HOST', '') %>
   email_sanitizer:
-    enabled: <%= ENV.fetch('EMAIL_SANITIZER_ENABLED', Rails.env.preview?) %>
-    to: <%= ENV.fetch('EMAIL_SANITIZER_TO', 'saniziter@example.com') %>
+    enabled: <%= ENV.fetch('EMAIL_SANITIZER_ENABLED', false) %>
```
Comment (akostadinov) [inline config/examples/settings.yml:35]: Also this seems not well controllable with ENV variable. I htink it makes sense to disable when env variable is empty.

File: config/secrets.yml [hunk 9]
```diff
@@ -17,28 +17,9 @@
...
-  <<: *base
-  secret_key_base: <%= ENV['SECRET_KEY_BASE'] %>
-
 # Do not keep production secrets in the unencrypted secrets file.
 # Instead, either read values from the environment.
 # Or, use `bin/rails secrets:setup` to configure encrypted secrets
 # and move the `production:` environment over there.
 production:
-  <<: *base
   secret_key_base: <%= ENV['SECRET_KEY_BASE'] %>
```
Comment (akostadinov) [inline config/secrets.yml:25]: This line sohuld also not be needed. Maybe remove the whole file?

File: openshift/system/Dockerfile [hunk 10]
```diff
@@ -70,7 +70,6 @@ RUN source /opt/app-root/etc/scl_enable \
 RUN chgrp root /opt/system/
 
 ADD . ./
-ADD config/docker/* ./config/
```
Comment (akostadinov) [inline openshift/system/Dockerfile:73]: we also need to change porta-midstream with this

File: openshift/system/config/cache_store.yml [hunk 11]
```diff
@@ -1,7 +1,3 @@
-base: &default
+production:
   - :mem_cache_store
-  - <%= ENV['MEMCACHE_SERVERS'].presence || 'system-memcache:11211' %>
-
-preview: *default
-
-production: *default
+  - <%= ENV.fetch('MEMCACHE_SERVERS', 'system-memcache:11211') %>
```
Comment (akostadinov) [inline openshift/system/config/cache_store.yml:3]: Original version seems better as it will ignore empty `MEMCACHE_SERVERS` variable. Or is this intentional?

File: config/examples/backend_redis.yml [hunk 12]
```diff
@@ -1,29 +1,17 @@
...
-  <<: *base
+  <<: *default
 
 test:
-  <<: *base
+  <<: *default
   db: <%= 2 + ENV['TEST_ENV_NUMBER'].to_i %>
+
+production:
+  <<: *default
```
Comment (akostadinov) [inline config/examples/backend_redis.yml:17]: Here also I'm a little concerned to leave production use same config as development as confusion may happen between the two environments if one starts production locally for whatever reasons. But since it was like that before, I guess we can do in another PR if we want.

File: config/application.rb [hunk 13]
```diff
@@ -175,12 +175,8 @@ def try_config_for(*args)
     config.three_scale.sandbox_proxy = ActiveSupport::OrderedOptions.new
     config.three_scale.sandbox_proxy.merge!(config_for(:sandbox_proxy).symbolize_keys)
 
-    config.three_scale.web_analytics = ActiveSupport::OrderedOptions.new
```
Comment (akostadinov) [inline config/application.rb:178]: On line 14 there is a reference to `preview` environment. I think we want it removed.

File: config/examples/oauth2.yml (hunk 6 above)
Comment (jlledom) [reply config/examples/oauth2.yml:3]: Setting `GITHUB_ENABLED = 0` ? Not sure if that would work

Comment (akostadinov) [reply config/examples/oauth2.yml:3]: 0 and "" are still truthy in ruby

**Comment by akostadinov** [comment]
One more thing, `config/environments/preview.rb` needs to be deleted I assume.

File: config/examples/backend_redis.yml (hunk 12 above)
Comment (mayorova) [reply config/examples/backend_redis.yml:17]: Yes, I also thought that actually, `production` environment probably doesn't have much sense in these example files.
The only reason I can think of is that if somebody wants to try the app locally in "production mode" (because some things change in the behavior).

Certainly we can change this later.

File: config/examples/redhat_customer_portal.yml [hunk 14]
```diff
@@ -1,19 +1,15 @@
...
+  entitlements_notifications_enabled: <%= ENV.fetch('RH_SUPPORT_ENTITLEMENTS_NOTIFICATIONS', '1') == '1' %>
   assign_entitlements_email: <%= ENV.fetch('RH_ASSIGN_ENTITLEMENTS_EMAIL', 'assign.entitlements@example.com') %>
-  revoke_entitlements_email: <%= ENV.fetch('RH_ASSIGN_ENTITLEMENTS_EMAIL', 'revoke.entitlements@example.com') %>
+  revoke_entitlements_email: <%= ENV.fetch('RH_REVOKE_ENTITLEMENTS_EMAIL', 'revoke.entitlements@example.com') %>
 
 development:
   <<: *default
-  client_id: 3scale
-  client_secret: replace-with-rhsso-client-secret-if-not-public
-  realm: http://localhost:8080/auth/realms/rh-customer-portal
```
[the remark this answers is not in the document]
Comment (mayorova) [reply config/examples/redhat_customer_portal.yml:15]: Well, it was never valid... I imagine it's just an example of how a URL of a RH-SSO realm looks like. We could keep `localhost`, but I think `sso.example.com` is just more explicit.

File: config/examples/redis.yml (hunk 7 above)
Comment (mayorova) [reply config/examples/redis.yml:17]: As I mentioned in [this comment](https://github.com/3scale/porta/pull/3225/files#r1123083017) I think having production environment is pretty useless for localhost in general, just when you want to test something which is production-specific. I don't think having a different DB will make much difference. But we can do it, just to follow the same pattern as for the relational DB.

Comment (mayorova) [reply config/examples/redis.yml:17]: Actually, I'm more inclined to delete the default value in `production` and make the developer set the value explicitly.

File: config/examples/settings.yml (hunk 8 above)
Comment (mayorova) [reply config/examples/settings.yml:35]: Right. I'll use the same approach as discussed [here](https://github.com/3scale/porta/pull/3225/files#r1123097194) for now, and we can change it later.

File: config/secrets.yml (hunk 9 above)
Comment (mayorova) [reply config/secrets.yml:25]: Sounds good :+1:

File: openshift/system/Dockerfile (hunk 10 above)
Comment (mayorova) [reply openshift/system/Dockerfile:73]: Yep!

File: openshift/system/config/cache_store.yml (hunk 11 above)
Comment (mayorova) [reply openshift/system/config/cache_store.yml:3]: Yes, good catch, you're right.

**Review by jlledom — APPROVED** [review]

File: config/examples/redis.yml (hunk 7 above)
Comment (akostadinov) [reply config/examples/redis.yml:17]: I think a good default is valuable. I had to start production locally a couple of times in the past. It's crap to need to touch many places instead of having things work mostly OOB

**Review by akostadinov — APPROVED** [review]
Except for the redis config where production doesn't use separate DB, I like the rest. We can have a new PR for that though.

File: Gemfile [hunk 15]
```diff
@@ -254,7 +254,7 @@ end
 gem 'webpacker', '5.4.4'
 
 gem 'developer_portal', path: 'lib/developer_portal'
-gem 'unicorn', require: false, group: %i[production preview]
```
Comment (josemigallas) [inline Gemfile:257]: Question: what about `staging`? Will there be no more distinction between that and production anymore?

File: config/examples/backend_redis.yml [hunk 16]
```diff
@@ -1,29 +1,18 @@
...
-# servers       list of servers (hostname:port) to connect to. The first one is the master, the
-#               others are used for failover (no sharding yet).
-# db            database to use. A number from 0 to 10 (default is 0)
-# timeout
-# password
-# thread_safe
-base: &base
-  url: "<%= ENV.fetch('BACKEND_REDIS_URL', 'redis://localhost:6379/0') %>"
+base: &default
+  url: "<%= ENV.fetch('BACKEND_REDIS_URL', 'redis://localhost:6379/6') %>"
```
Comment (josemigallas) [inline config/examples/backend_redis.yml:2]: The URL has change slightly, is it intentional?

File: config/examples/redis.yml (hunk 7 above)
Comment (josemigallas) [reply config/examples/redis.yml:17]: In case we remove the default values, please @mayorova add any necessary extra steps to the docs

File: config/examples/rolling_updates.yml [hunk 17]
```diff
@@ -14,22 +13,17 @@ base: &default
   policy_registry: true
   proxy_private_base_path: true
   service_mesh_integration: false
-  api_as_product: false
+  apicast_v2: true
```
Comment (josemigallas) [inline config/examples/rolling_updates.yml:16]: Could we sort these alphabetically?

File: lib/three_scale/domain_substitution.rb [hunk 18]
```diff
@@ -12,7 +12,7 @@ def internal_host
   end
 
   # == Context
-  # In any debugging environment _development_ or _preview_,
+  # In any debugging environment. e.g. _development_,
```
Comment (josemigallas) [inline lib/three_scale/domain_substitution.rb:15]: Is there any other debugging env?
```suggestion
  # In any debugging environment, i.e. _development_,
```

**Review by josemigallas — COMMENTED** [review]
LGTM although it's kinda hard to review

File: Gemfile (hunk 15 above)
Comment (mayorova) [reply Gemfile:257]: There is no distinction indeed. Staging uses `Rails.env == 'production'`

The only difference is `BUGSNAG_RELEASE_STAGE`, which is set to `staging` in staging, to distinguish the bug reports in BugSnag.

File: config/examples/backend_redis.yml (hunk 16 above)
Comment (mayorova) [reply config/examples/backend_redis.yml:2]: Yes, this is intentional, and the reason is that [porta-dev-tools](https://github.com/3scale-labs/porta-dev-tools/blob/master/bin/porta#L705) uses DB

File: config/examples/redis.yml (hunk 7 above)
Comment (mayorova) [reply config/examples/redis.yml:17]: I think it makes it more confusing if we reserve all different DB numbers for different environments. But I agree that it probably needs to be documented.

File: config/examples/rolling_updates.yml (hunk 17 above)
Comment (mayorova) [reply config/examples/rolling_updates.yml:16]: I have this in the PR description:

> P.S. I also had an idea to reorder the keys in the config files in the alphabetical order, but not sure how practical this is. To me it seems better to have them sorted, but if we decide to do so, I'd do it in another PR, otherwise it's impossible to compare the changes.

It's already not easy to track the changes to be honest :upside_down_face: 
But reordering would make it 100% impossible. So, yes, will do in another PR.

**Review by akostadinov — DISMISSED** [review]

**Review by akostadinov — APPROVED** [review]
:partying_face:
